src/v/utils/uuid.h

```diff
@@ -0,0 +1,90 @@
+// Copyright 2022 Redpanda Data, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.md
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
+#pragma once
+
+#include "bytes/details/out_of_range.h"
+#include "reflection/adl.h"
+
+#include <absl/hash/hash.h>
+#include <boost/functional/hash.hpp>
+#include <boost/uuid/uuid.hpp>
+#include <boost/uuid/uuid_io.hpp>
+
+#include <vector>
+
+// Wrapper around Boost's UUID type suitable for serialization with serde.
+// Provides utilities to construct and convert to other types.
+//
+// Expected usage is to supply a UUID v4 (i.e. based on random numbers). As
+// such, serialization of this type simply serializes 16 bytes.
+struct uuid_t {
```

Comment:
is there an opportunity / is it possible to unify this with the uuid implementation in `src/v/kafka/types.h`? i noticed that it is uses `using underlying_t = std::array<uint8_t, length>;` which might also make stuff like hashing (`for (const uint8_t byte : u.uuid) {` below...) more efficient?

Reply:
EDIT: i didn't read your PR cover letter and now i see

> Note that there already exists a kafka::uuid type that wraps std::array<uint18_t, 16>. I opted to avoid reusing it since it is exposed via the Kafka API, making it difficult to modify in the future if need be.

makes sense, and I generally think it is a good thing to keep `kafka::*` isolated.

Reply:
@dotnwat it may make sense to have a base type in `model::*` and extend it in 2 places ? i think that's what scylla did at some point if my memory serves me well.

Reply:
makes sense